helpers/misc.py

```diff
@@ -352,7 +352,7 @@ def remove_overlap_poly(gdf_temp, id_to_keep):
                         rsuffix="right")
                 
     # Remove geometries that intersect themselves and duplicates
-    gdf_temp = gdf_temp[gdf_temp.index > gdf_temp.index_right].copy()
+    gdf_temp = gdf_temp[gdf_temp.index != gdf_temp.index_right].copy()
```

Comment:
I've reverted back to the initial state because it wasn't working properly using ">". At this stage, it's okay to have the same geometry in the geom1 and geom2 columns, we want to keep them all and the filter is done after. We want to identify and group overlapping polygons that are most likely a detection of a single object. In the groups, only one polygon is identified, and the one with the highest confidence score is retained. Using “>” does not perform this selection.